src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -117,6 +170,22 @@ def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
                     code="invalid",
                 )
 
+        # Make sure we have a variable-eigenshap mapping
+        if variables_properties := attrs.get("variables_properties"):
+            for variable_property in variables_properties:
+                component_key = variable_property.get("component_key")
+                property = variable_property.get("eigenshap")
+
+                if (component_key and not property) or (property and not component_key):
+                    raise serializers.ValidationError(
+                        {
+                            "variables_properties": _(
+                                "Both variable selection and property name are required."
+                            )
+                        },
+                        code="invalid",
+                    )
```

Comment:
Can't we make `component_key` and `eigenshap` both required in `MappedVariablePropertySerializer`? Or is there a use case when both can be empty?

Reply:
I agree that it would make sense to make both fields required in the `MappedVariablePropertySerializer`. In addition, I think we should add a check like the one of the roltype below:
If the form designer wants to relate a variable to an eigenschap, we need to make sure that the eigenschap has an eigenschaptype compatible (related) to the zaaktype specified.